File: typescript/packages/express/examples/basic-server.js
/**
 * Basic Express Server Example with Prism Payment Middleware
 * 
 * This example demonstrates how to set up a simple Express server with payment-protected routes.
 */

const express = require('express');
const { prismPaymentMiddleware } = require('@financedistrict/prism-x402-sdk-express');

const app = express();

// Configure Prism middleware
const paymentMiddleware = prismPaymentMiddleware(
  {
    // API Configuration
    apiKey: process.env.PRISM_API_KEY || 'dev-key-123',
    baseUrl: process.env.PRISM_BASE_URL || 'https://prism-gw.test.1stdigital.tech',
    
    // Enable debug mode to see detailed logs
    debug: true  // Set to false in production
  },
  {
    // Protected Routes
    '/api/premium': {
      price: 0.01, // 0.01 ETH
      description: 'Premium API access',
      mimeType: 'application/json'
    },
    '/weather': {
      price: '$0.001', // $0.001 USD worth of ETH
      description: 'Weather data access'
    },
    '/data/*': {
      price: 0.005, // Wildcard: matches /data/anything
      description: 'Data API access'
    }
  }
);

// Apply Prism payment middleware globally
app.use(paymentMiddleware);

// Free routes (no payment required)
app.get('/', (req, res) => {
  res.send(`
    <html>
      <head><title>Prism Payment Demo</title></head>
      <body>
        <h1>Prism Express Payment Demo</h1>
        <p>This is a free route. Try these payment-protected routes:</p>
        <ul>
          <li><a href="/api/premium">/api/premium</a> - Premium API (0.01 ETH)</li>
          <li><a href="/weather">/weather</a> - Weather data ($0.001)</li>
          <li><a href="/data/users">/data/users</a> - Data API (0.005 ETH)</li>
        </ul>
        <p>Without payment, you'll receive a 402 Payment Required response with payment requirements.</p>
      </body>
    </html>
  `);
});

// Protected routes (payment verified by middleware)
app.get('/api/premium', (req, res) => {
  res.json({
    message: 'Welcome to Premium API!',
    payer: req.payer, // Payer address (set by middleware)
    data: {
      premium: true,
      features: ['Advanced analytics', 'Priority support', 'Custom integrations']
    }
  });
});

app.get('/weather', (req, res) => {
  res.json({
    location: 'San Francisco',
    temperature: 72,
    condition: 'Sunny',
    humidity: 65,
    payer: req.payer
  });
});

app.get('/data/:resource', (req, res) => {
  const resource = req.params.resource;
  res.json({
    resource,
    data: ['item1', 'item2', 'item3'],
    payer: req.payer
  });
});

// Start server
const PORT = process.env.PORT || 3000;
app.listen(PORT, () => {
  console.log(`Server running on http://localhost:${PORT}`);
  console.log('Routes:');
  console.log('  GET /              - Free route');
  console.log('  GET /api/premium   - Requires payment (0.01 ETH)');
  console.log('  GET /weather       - Requires payment ($0.001)');
  console.log('  GET /data/*        - Requires payment (0.005 ETH)');
});

File: python/packages/flask/examples/basic_server.py
"""
Basic Flask Server Example with Prism Payment Middleware
"""

from flask import Flask, g, jsonify
from prism_flask import prism_payment_middleware
from prism_sdk_core import PrismMiddlewareConfig, RoutePaymentConfig

app = Flask(__name__)

# Configure Prism middleware
prism_payment_middleware(
    app,
    config=PrismMiddlewareConfig(
        api_key='dev-key-123',
        base_url='https://prism-gw.test.1stdigital.tech',
        debug=True
    ),
    routes={
        '/api/premium': RoutePaymentConfig(
            price=0.01,
            description='Premium API access'
        ),
        '/api/weather': RoutePaymentConfig(
            price='$0.001',
            description='Weather data access'
        )
    }
)


@app.route('/')
def home():
    """Free route - no payment required"""
    return '''
    <html>
        <head><title>Prism Flask Demo</title></head>
        <body>
            <h1>Prism Flask Payment Demo</h1>
            <p>This is a free route. Try these payment-protected routes:</p>
            <ul>
                <li><a href="/api/premium">/api/premium</a> - Premium API (0.01 ETH)</li>
                <li><a href="/api/weather">/api/weather</a> - Weather data ($0.001)</li>
            </ul>
            <p>Without payment, you'll receive a 402 Payment Required response.</p>
        </body>
    </html>
    '''


@app.route('/api/premium')
def premium():
    """Payment-protected premium endpoint"""
    payer = getattr(g, 'prism_payer', None)
    payment = getattr(g, 'prism_payment', None)
    
    return jsonify({
        'message': 'Welcome to Premium API!',
        'payer': payer,
        'payment': payment,
        'data': {
            'premium': True,
            'features': [
                'Advanced analytics',
                'Priority support',
                'Custom integrations'
            ]
        }
    })


@app.route('/api/weather')
def weather():
    """Payment-protected weather endpoint"""
    payer = getattr(g, 'prism_payer', None)
    
    return jsonify({
        'location': 'San Francisco',
        'temperature': 72,
        'condition': 'Sunny',
        'humidity': 65,
        'wind': '10 mph',
        'forecast': [
            {'day': 'Monday', 'temp': 70, 'condition': 'Partly Cloudy'},
            {'day': 'Tuesday', 'temp': 68, 'condition': 'Cloudy'},
            {'day': 'Wednesday', 'temp': 73, 'condition': 'Sunny'}
        ],
        'payer': payer
    })


if __name__ == '__main__':
    print('Server running on http://localhost:5000')
    print('Routes:')
    print('  GET /              - Free route')
    print('  GET /api/premium   - Requires payment (0.01 ETH)')
    print('  GET /api/weather   - Requires payment ($0.001)')
    app.run(debug=True, port=5000)
